specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/computeRPCommon.json

```diff
@@ -1913,6 +1913,89 @@
         }
       },
       "description": "Specifies additional settings to be applied when patch mode AutomaticByPlatform is selected in Linux patch settings."
+    },
+    "AttachDetachDataDisksRequest": {
+      "properties": {
+        "attachDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AttachDataDisk"
+          },
+          "description": "The list of managed data disks to be attached."
+        },
+        "detachDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DetachDataDisk"
+          },
+          "description": "The list of managed data disks to be detached."
+        }
+      },
+      "description": "Specifies the input for attaching and detaching a list of managed data disks."
+    },
+    "AttachDetachDataDisksResponse": {
+      "properties": {
+        "attachedDataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AttachedDataDisk"
+          },
+          "description": "The list of managed data disks that were attached."
+        }
+      },
+      "description": "Response for attach and detach data disks request."
+    },
+    "AttachDataDisk": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ID of the managed data disk."
+        },
+        "lun": {
+          "type": "integer",
+          "description": "The logical unit number of the data disk. This value is used to identify data disks within the VM and therefore must be unique for each data disk attached to a VM. If not specified, lun would be auto assigned."
+        }
+      },
+      "required": [
+        "id"
+      ],
+      "description": "Describes the data disk to be attached."
+    },
+    "DetachDataDisk": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "ID of the managed data disk."
+        },
+        "detachOption": {
+          "$ref": "#/definitions/DetachOption",
+          "description": "Specifies the detach behavior to be used while detaching a disk or which is already in the process of detachment from the virtual machine. Supported values: **ForceDetach**. <br><br> detachOption: **ForceDetach**. If a previous detachment attempt of the data disk did not complete due to an unexpected failure from the virtual machine and the disk is still not released then use force-detach as a last resort option to detach the disk forcibly from the VM. All writes might not have been flushed when using this detach behavior. <br><br> This feature is still in preview mode and is not supported for VirtualMachineScaleSet."
```

Comment:
**init: [suggestion]** description seems to be duplicate one mentioned for DetachOption object. We can just specify "Supported options available for Detach of a disk from a VM. Refer to DetachOption reference for more details".